client/cli/src/lib.rs

```diff
@@ -982,3 +991,40 @@ impl FedimintCli {
         }
     }
 }
+
+/// Convert clap arguments to backup metadata
+fn metadata_from_clap_cli(metadata: Vec<String>) -> Result<BTreeMap<String, String>, CliError> {
+    let metadata: BTreeMap<String, String> = metadata
+        .into_iter()
+        .map(|item| {
+            match &item
+                .splitn(2, '=')
+                .map(ToString::to_string)
+                .collect::<Vec<String>>()[..]
+            {
+                [] => Err(anyhow::format_err!("Empty metadata argument not allowed")),
+                [key] => Err(anyhow::format_err!("Metadata {key} is missing a value")),
+                [key, val] => Ok((key.clone(), val.clone())),
+                [..] => unreachable!(),
+            }
+        })
+        .collect::<anyhow::Result<_>>()
+        .map_err_cli_msg(CliErrorKind::GeneralFederationError, "failed")?;
```

Comment:
`CliErrorKind::InvalidValue` error would be better here I think?

and maybe extend `failed` to `parsing backup metadata failed`? If I understand that correctly we have the other error message regarding what exactly went wrong with the metadata argument as well, but I think it'd be nice if that error message would be a bit more descriptive